Userland/Services/WebServer/main.cpp

```diff
@@ -15,16 +15,25 @@
 
 int main(int argc, char** argv)
 {
+    String default_listen_address = "0.0.0.0";
     u16 default_port = 8000;
     const char* root_path = "/www";
 
+    String listen_address = default_listen_address;
     int port = default_port;
 
     Core::ArgsParser args_parser;
+    args_parser.add_option(listen_address, "IP address to listen on", "listen-address", 'l', "listen_address");
     args_parser.add_option(port, "Port to listen on", "port", 'p', "port");
     args_parser.add_positional_argument(root_path, "Path to serve the contents of", "path", Core::ArgsParser::Required::No);
     args_parser.parse(argc, argv);
 
+    auto ipv4_address = IPv4Address::from_string(listen_address);
+    if (!ipv4_address.has_value()) {
+        printf("Warning: invalid listen address: %s\n", listen_address.characters());
+        ipv4_address = IPv4Address::from_string(default_listen_address);
```

Comment:
That seems "dangerous" - passing an invalid IPv4 address should not make the webserver listen on 0.0.0.0, just print to stderr instead and exit.

```suggestion
        warnln("Invalid listen address: {}", listen_address);
        return 1;
```

Reply:
Yes good one, I fixed it.

Reply:
I just noticed we do the same for the port, I'll change that separately. Programs trying to be "helpful" that way isn't helping at all, IMO :D